kbncontent.go

```diff
@@ -0,0 +1,248 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+// Package kbncontent implements routines for analyzing Kibana content.
+//
+// It provides information about Kibana assets and a single source of truth for what things are legacy and should no longer be used.
+package kbncontent
+
+import (
+	"encoding/json"
+
+	// TODO - consider fully replacing jsonpath with objx
+	"github.com/PaesslerAG/jsonpath"
+	"github.com/stretchr/objx"
+)
+
+// Describes a visualization.
+// In this case, "visualization" means anything which can be embedded in a dashboard.
+type VisDesc struct {
+	Doc    map[string]interface{}
+	SoType string
+	Link   string
+
+	// root-level visualization type
+	// currently empty for Lens
+	Type string
+
+	// TSVB visualizations are always type "metrics"
+	// this property gives the TSVB sub type (gauge, markdown, etc)
+	TSVBType string
+
+	// meant to be a visualization-editor-agnostic name for what
+	// kind of visualization this actually is (pie, bar, etc)
+	// Note: does not yet support Lens
+	SemanticType string
+
+	Title    string
+	IsLegacy bool
+}
+
+func isLegacy(soType, visType string) bool {
+	return soType == "visualization" && visType != "markdown" && visType != "input_control_vis" && visType != "vega"
```

Comment:
Nit. Separate this in different conditions for readability and easier extendability.
```suggestion
	if soType == "visualization" {
		switch visType {
		case "markdown", "input_control_vis", "vega":
			return false
		}
	}
	return true
```